tests/evhtests/l7_evh_crd_test.go

```diff
@@ -1869,3 +1869,445 @@ func TestGoodToBadSSORuleForEvh(t *testing.T) {
+		RemoveListeningPortOnVsDown:   true,
+		SslSessCacheAvgSize:           2024,
+	}.L7Rule()
+	l7rule2.ResourceVersion = "2"
+	_, err = v1alpha2CRDClient.AkoV1alpha2().L7Rules("default").Update(context.TODO(), l7rule2, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating L7Rule: %v", err)
+	}
+	g.Eventually(func() string {
+		l7Rule, _ := v1alpha2CRDClient.AkoV1alpha2().L7Rules("default").Get(context.TODO(), l7ruleName2, metav1.GetOptions{})
+		return l7Rule.Status.Status
+	}, 10*time.Second).Should(gomega.Equal("Accepted"))
+
+	//Update hostrule with L7rule2
+	hrUpdate = integrationtest.FakeHostRule{
+		Name:      hrname,
+		Namespace: "default",
+		Fqdn:      "foo.com",
+	}.HostRule()
+	hrUpdate.Spec.VirtualHost.L7Rule = l7ruleName2
+	hrUpdate.ResourceVersion = "3"
+	_, err = v1beta1CRDClient.AkoV1beta1().HostRules("default").Update(context.TODO(), hrUpdate, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating HostRule: %v", err)
+	}
+
+	g.Eventually(func() string {
+		hostrule, _ := v1beta1CRDClient.AkoV1beta1().HostRules("default").Get(context.TODO(), hrname, metav1.GetOptions{})
+		return hostrule.Status.Status
+	}, 10*time.Second).Should(gomega.Equal("Accepted"))
+	g.Eventually(func() bool {
+		_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+		if len(nodes) > 0 {
+			return *nodes[0].EvhNodes[0].AllowInvalidClientCert
+		}
+		return true
+	}, 10*time.Second).Should(gomega.Equal(false))
+	_, aviModel = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+	g.Expect(*nodes[0].EvhNodes[0].RemoveListeningPortOnVsDown).To(gomega.Equal(true))
+	g.Expect(*nodes[0].EvhNodes[0].CloseClientConnOnConfigUpdate).To(gomega.Equal(true))
+	g.Expect(*nodes[0].EvhNodes[0].AllowInvalidClientCert).To(gomega.Equal(false))
+	g.Expect(*nodes[0].EvhNodes[0].IgnPoolNetReach).To(gomega.Equal(false))
+	g.Expect(*nodes[0].EvhNodes[0].BotPolicyRef).To(gomega.ContainSubstring("thisisaviref-botpolicy"))
+	g.Expect(*nodes[0].EvhNodes[0].SslSessCacheAvgSize).To(gomega.Equal(uint32(2024)))
+	g.Expect(*nodes[0].EvhNodes[0].MinPoolsUp).To(gomega.Equal(uint32(0)))
+	g.Expect(nodes[0].EvhNodes[0].HostNameXlate).To(gomega.BeNil())
+	g.Expect(nodes[0].EvhNodes[0].SecurityPolicyRef).To(gomega.BeNil())
+	integrationtest.TeardownHostRule(t, g, sniVSKey, hrname)
+	if err := lib.AKOControlConfig().V1alpha2CRDClientset().AkoV1alpha2().L7Rules("default").Delete(context.TODO(), l7ruleName, metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("error in deleting l7Rule: %v", err)
+	}
+	if err := lib.AKOControlConfig().V1alpha2CRDClientset().AkoV1alpha2().L7Rules("default").Delete(context.TODO(), l7ruleName2, metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("error in deleting l7Rule: %v", err)
+	}
+	TearDownIngressForCacheSyncCheck(t, modelName)
+}
+
+func TestValidToInvalidL7rule(t *testing.T) {
```

Comment:
Should we have test cases for 
1. PArent VS related properties
2. If hostrule deleted and L7Rule kept as it is, what is behaviour.

Reply:
added test case for dedicated vs and the 2nd point. will add for parent VS in a seperate Pr